src/Disks/ObjectStorages/DiskObjectStorage.cpp

```diff
@@ -724,7 +724,7 @@ std::unique_ptr<ReadBufferFromFileBase> DiskObjectStorage::readFile(
         && read_settings.enable_filesystem_cache;
 
     size_t buffer_size = prefer_bigger_buffer_size
-        ? std::max<size_t>(settings.remote_fs_buffer_size, DBMS_DEFAULT_BUFFER_SIZE)
+        ? std::max<size_t>(settings.remote_fs_buffer_size, settings.prefetch_buffer_size)
```

Comment:
I am not sure, this setting was not intended for this exactly, but as `prefetch_buffer_size` is anyway DBMS_DEFAULT_BUFFER_SIZE by default and never changed unless user changes it, then I assume ok as it will allow to make this behaviour changeable. I did notice some time ago that we no longer use prefetch_buffer_size setting anyhow, but did not see a need to revive it in code really (should have removed, but I did not), anyway this revival from this PR looks ok. One thing - probably it should be renamed? As it is not really a prefetch buffer size, but just buffer size used for both prefetch and read.

Reply:
> it can be used only in case of prefer_bigger_buffer_size

Yes, but it is enabled by default :)